sdk/core/azure-core/vcpkg/vcpkg.json

```diff
@@ -22,6 +22,10 @@
     {
       "name": "vcpkg-cmake-config",
       "host": true
+    },
+    {
+      "name": "wil",
+      "platform": "windows"
```

Comment:
I saw this change went into main as well, but wanted to ask:
Why do we need this dependency (https://github.com/microsoft/wil)?

Reply:
Wil is an RAII wrapper library authored by the Windows team to provide RAII wrappers for Windows types. Because there were objections to the RAII wrappers I wrote, I needed to find an RAII wrapper for 3 Bcrypt handle types, and wil is a well tested supported library.

I tried to use an RAII wrapper entirely within Azure and the STL, but there was a disagreement about naming and adopting WIL allowed me to make forward progress with the PR.